tests/e2e/csi_cns_telemetry.go

```diff
@@ -0,0 +1,207 @@
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+)
+
+var _ = ginkgo.Describe("[csi-block-vanilla] [csi-file-vanilla] CNS-CSI Cluster Distribution Telemetry", func() {
+	f := framework.NewDefaultFramework("csi-cns-telemetry")
+	var (
+		client       clientset.Interface
+		namespace    string
+		scParameters map[string]string
+		datastoreURL string
+	)
+
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		scParameters = make(map[string]string)
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		//Reset the cluster distribution value to original value
+		isSet := setClusterDistributionValue4Vanilla(ctx, client, vanillaClusterDistribution)
+		gomega.Expect(isSet).To(gomega.BeTrue())
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		//Reset the cluster distribution value to original value
+		isSet := setClusterDistributionValue4Vanilla(ctx, client, vanillaClusterDistribution)
+		gomega.Expect(isSet).To(gomega.BeTrue())
+	})
+
+	/*
+		Test to verify cluster distribution set in PVC is being honored during volume creation.
+		Steps
+			1. Create StorageClass
+			2. Create PVC with valid disk size
+			3. Expect PVC to pass and cluster distribution value set
+			4. Update cluster-distribution value
+			5. Create another PVC with valid disk size
+			6. Expect PVC to pass and cluster distribution value set to latest update
+			7. Expect the old PVC to reflect the latest cluster-distribution value
+	*/
+
+	// Test for cluster-distribution value presence
+	ginkgo.It("Verify dynamic provisioning of pvc has cluster-distribution value updated", func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ginkgo.By("Invoking cluster-distribution for volume basic test")
+		var storageclasspvc1 *storagev1.StorageClass
+		var pvclaim1 *v1.PersistentVolumeClaim
+		var err error
+
+		scParameters[scParamDatastoreURL] = datastoreURL
+		storageclasspvc1, pvclaim1, err = createPVCAndStorageClass(client, namespace, nil, scParameters, diskSize, nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc1.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		persistentvolumes1, err := fpv.WaitForPVClaimBoundPhase(client, []*v1.PersistentVolumeClaim{pvclaim1}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
```

Comment:
Can you print the err here?

Reply:
gomega take cares of printing the error details on the failure